js/liqui.js

```diff
@@ -152,7 +161,7 @@ module.exports = class liqui extends Exchange {
                 'price': priceLimits,
                 'cost': costLimits,
             };
-            let active = (market['hidden'] == 0);
+            let active = (market['hidden'] === 0);
```

Comment:
Did you check every property if they are actually integers, or you just simply changing them to be strict?
Because these tests could [cause bugs](https://github.com/ccxt/ccxt/commit/816004216bbac692f779770c72ef8613b4041bf1#r26845226).

Reply:
Yeah, I see. Just checked - they return numbers.

Reply:
This approach of manual checking actually bears a great amount of cognitive load, as for me.

If this is a code, it should be a 'number'.

I just think if it's possible to create a schema and validate against it automatically and then automatically convert 'strings' to 'numbers' when we expect a number.

Reply:
> I just think if it's possible to create a schema and validate against it automatically and then automatically convert 'strings' to 'numbers' when we expect a number.

I tried answering the schema question here: https://github.com/ccxt/ccxt/issues/17#issuecomment-314935916

In short: I don't think we can really design such a **universally-good fast and practical schema**, unfortunately... That many differences would demand a lot of special handling and would render the whole effort useless in the end, I think. Adding WS would make the abstraction layer even more complicated than REST, there's a lot more that can go wrong in streaming mode. I'm still open to a possibility of making a unified schema, though.

Reply:
You don't need to check it, simply use `safeInteger` or `safeString`. But, if you don't use any of these (which is mostly equivalent to `safeValue`) you have to check. You didn't, that's why I mentioned.

I also didn't say if it's a code then it should be a number, I said `code`, which was the property name there.

My point was there; if you want to bypass these `===` problems with numeric values (int or string), then simply use `safeInteger` (or `safeString`, but then obviously your comparisons changes too).

Reply:
By 'checking' I mean checking if incoming value is wrapped into `safe*` family, not wrapping a value itself. It could be easily missed, it **is** gets missed an I often see commits targeted at fixing those missing safe wraps.

Reply:
Aha, I misunderstood your comment then. However developing schema system seems even more work (as discussed in the issue kroitor linked)...

Reply:
Agree.